10up/classifai
# feat/115: Add Microsoft Azure Text to speech capability

### Description of the Change



Closes #115

### How to test the Change

1. Visit `/wp-admin/admin.php?page=language_processing&tab=azure_text_to_speech`.
2. Setup the endpoint and key. 
3. Select post types which should support this feature.
4. Select a voice.

<img width="1279" alt="Screenshot 2023-03-31 at 2 20 41 PM" src="https://user-images.githubusercontent.com/17757960/229073687-cfd58e34-9d7c-437a-9c8a-ca67c6c4425d.png">

5. Add a new Post/Page, whichever has support for the feature.
6. The feature will be "on" by default if the post type supports it.

<img width="291" alt="Screenshot 2023-03-31 at 2 22 37 PM" src="https://user-images.githubusercontent.com/17757960/229073904-7e162f80-f7f3-4a06-8ead-b2f3bda599ea.png">

7. Save the post and 

![classifai mylocal_wp-admin_post php_post=581 action=edit](https://user-images.githubusercontent.com/17757960/229074442-a7ca1b3b-70e5-4593-855e-8714bf2b2591.png)

8. New controls such as **Refresh** and **Preview** will be visible. Refresh can be used if you decide to change the voice.
9. Go to the frontend, the post will have a play button added using `the_content` filter.


<img width="713" alt="Screenshot 2023-03-31 at 2 26 14 PM" src="https://user-images.githubusercontent.com/17757960/229074842-b98494f9-ae8b-466c-b4b4-ab03928b1a75.png">


### Changelog Entry

> Added - Text to Speech capability using Microsoft Azura Text to Speech 

### Credits

Props @jeffpaul @Sidsector9 


### Checklist:
